sidecar/src/xline.rs

```diff
@@ -88,17 +105,97 @@ impl XlineHandle {
     }
 
     /// Start the xline server
-    pub(crate) fn start(&self) -> Result<()> {
-        // TODO start xline server
+    pub(crate) async fn start(&mut self) -> Result<()> {
+        // TODO: hold a distributed lock during start
+
+        // Step 1: Check if there is any node running
+        // Step 2: If there is no node running, start single node cluster
```

Comment:
Why start a single node cluster here, currently it seems Xline does not support membership change for a single node cluster, single node is designed for test only.

Reply:
We should support it because etcd also supports it.